File: split/triefix.pas
procedure triefix(p: triepointer);
var
  q: triepointer;
  c: ASCIIcode;
  z: triepointer;
begin
  z := triehash[p];
  repeat
    q := triel[p];
    c := triec[p];
    trie[z + c].rh := triehash[q];
    trie[z + c].b1 := c + 0;
    trie[z + c].b0 := trieo[p];
    if q > 0 then
      triefix(q);
    p := trier[p];
  until p = 0;
end;

File: cpp/triefix.cpp
#include "triefix.h"

void triefix(triepointer p)
{
	triepointer z = triehash[p];
	do
	{
		triepointer q = triel[p];
		ASCIIcode c = triec[p];
		trie[z+c].rh = triehash[q];
		trie[z+c].b1 = c;
		trie[z+c].b0 = trieo[p];
		if (q > 0)
			triefix(q);
		p = trier[p];
	} while (p);
}
